File: frontend/src/app/sites/catan/catan.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {GameState} from "./dto/game/GameState";
import {GameConfig} from "./dto/game/GameConfig";
import {GameSessionService} from "../all/services/http/game-session.service";
import {environment} from "../../../environments/environment";
import {NotificationService} from "../../main/core/services/notification.service";

@Injectable({
    providedIn: 'root'
})
export class CatanService extends GameSessionService<GameConfig, GameState> {
    protected readonly baseUri: string = environment.httpApiUrl + '/catan';

    constructor(httpClient: HttpClient, notification: NotificationService) {
        super(httpClient, notification);
    }

    rollDice(isAlchemist: boolean): Observable<GameState> {
        return this.httpClient.post<GameState>(this.baseUri + '/dice-roll', {},
            {
                params: {
                    alchemist: isAlchemist
                }
            });
    }

    undoRoll(): Observable<GameState> {
        return this.httpClient.post<GameState>(this.baseUri + '/undo', {});
    }
}

File: backend/src/main/java/net/zorphy/backend/site/catan/service/CatanService.java
package net.zorphy.backend.site.catan.service;

import net.zorphy.backend.site.core.http.service.GameSessionSaveService;
import net.zorphy.backend.site.core.http.dto.ResultState;
import net.zorphy.backend.site.catan.dto.game.GameConfig;
import net.zorphy.backend.site.catan.dto.game.GameState;

public interface CatanService extends GameSessionSaveService<GameConfig, GameState, ResultState> {
    GameState rollDice(GameState oldState, boolean isAlchemist);

    GameState undoRoll(GameState oldState);
}